apps/authoring-services/src/main/java/com/infosys/lexauthoringservices/serviceimpl/CompetencyServiceImp.java

```diff
@@ -0,0 +1,146 @@
+package com.infosys.lexauthoringservices.serviceimpl;
+
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.fasterxml.jackson.databind.SerializationFeature;
+import com.infosys.lexauthoringservices.model.Competency;
+import com.infosys.lexauthoringservices.producer.Producer;
+import com.infosys.lexauthoringservices.service.CompetencyService;
+import com.infosys.lexauthoringservices.service.ContentCrudService;
+import com.infosys.lexauthoringservices.util.LexLogger;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.stereotype.Service;
+import org.springframework.util.ObjectUtils;
+import org.springframework.web.client.HttpClientErrorException;
+import org.springframework.web.client.RestTemplate;
+
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+@Service
+public class CompetencyServiceImp implements CompetencyService {
+
+    @Autowired
+    private LexLogger logger;
+
+    @Autowired
+    private Producer producer;
+
+    @Autowired
+    private ContentCrudService contentCrudService;
+
+    @Autowired
+    private GraphServiceImpl graphService;
+
+    @Autowired
+    private RestTemplate restTemplate;
+
+    @Value("${kafka.topics.competency.update}")
+    private String competencyUpdateTopic;
+
+    @Value("${sbext.service.host}")
+    private String sbextServiceURL;
+
+    @Value("${sbext.service.competencySearchPath}")
+    private String competencySearchPath;
+
+    public final String COMPETENCIES_CONST = "competencies";
+
+    public final String ID_CONST = "id";
+
+    public final String NAME_CONST = "name";
+
+    public final String DESCRIPTION_CONST = "description";
+
+    public final String IDENTIFIER_CONST = "identifier";
+
+    public final String TYPE_CONST = "type";
+
+    @Override
+    public void updateCompetencyData(Competency competency) {
+        producer.push(competencyUpdateTopic, competency);
+    }
+
+    /**
+     * Update the copetency in neo4j & elasticsearch
+     *
+     * @param competency
+     * @return
+     */
+    @Override
+    public Competency processUpdateCompetencyData(Competency competency) {
+        StringBuilder builder = new StringBuilder();
+        String path = competencySearchPath.replace("{org}", competency.getRootOrg()).replace("{id}", competency.getId());
+        builder.append(sbextServiceURL).append(path);
+        Object response = fetchResult(builder);
+        if (ObjectUtils.isEmpty(response))
+            return competency;
+        List<HashMap<String, Object>> listOfIds = (List<HashMap<String, Object>>) response;
+        if (!listOfIds.isEmpty()) {
```

Comment:
instead of negation, just return if it is empty... 
if(listOfIds.isEmpty()) {
    //log...
    return
}